const BaseCommand = require("../../../structures/BaseSlashCommand");

module.exports = class UnbanCmd extends BaseCommand {
  constructor(client) {
    super({
      name: "unban",
      description: "[ üõ°Ô∏è Modera√ß√£o ] Retire o banimento de algu√©m com esse comando."
    }, {
      clientPerms: ["banMembers"],
      userPerms: ["banMembers"]
    }, [{
      type: 6,
      name: "usu√°rio",
      description: "O usu√°rio que ter√° o seu banimento retirado",
      required: true
    }]);
    this.client = client;
  }
  
  async exec(ctx) {
    try {
      const userid = ctx.options.find(n => n.name == "usu√°rio").value;
      let user;
      
      try {
        user = await this.client.getRESTUser(userid);
      } catch(_) {
        await this.handleCustomError(ctx, "Usu√°rio n√£o encontrado!", { ephemeral: true, emoji: "‚ùì"});
      }
      
      if(user) {
        let bans = await ctx.guild.getBans();
        let find = bans.find(n => n.user.id == user.id);
        if(!find) {
          return await this.handleCustomError(ctx, "Esse usu√°rio n√£o est√° banido.", 1, { ephemeral: true });
        }
        try {
          await ctx.guild.unbanMember(userid);
          await this.handleSuccess(ctx, `O banimento de: \`${user.username}\` foi retirado!`, 1);
        } catch(_) {
          await this.handleCustomError(ctx, `Houve um erro inesperado ao retirar o banimento: \`${_}\``, 1, { ephemeral: true });
          console.log(_);
        }
      }
    } catch(_) {
      console.log(_);
      await this.handleError(ctx);
    }
  }
}